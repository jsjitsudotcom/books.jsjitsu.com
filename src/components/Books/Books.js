import React, { PureComponent } from "react";
import Style from "./Books.scss";
import PropTypes from "prop-types";

import Connect from "./containers/Books";

import Book from "./../Book/Book";

const Loading = Array.from(new Array(4)).map((e, i) => (
  <Book key={i + "/loading"} loading />
));

class Books extends PureComponent {
  render() {
    return (
      <div className={Style.container}>
        {this.props.books.length === 0 && !this.props.loading && (
          <div className={Style.message}>
            Aucun livre trouv√©, veuillez changer votre recherche
          </div>
        )}
        <div className={Style.books}>
          {this.props.books
            .map((book, i) => (
              <div className={Style.book} key={i}>
                <Book
                  title={book.title}
                  date={book.date}
                  author={book.author}
                  cover={book.cover}
                />
              </div>
            ))
            .concat(this.props.loading ? Loading : [])}
        </div>
      </div>
    );
  }
}

Books.propTypes = {
  books: PropTypes.array
};

Books.defaultProps = {
  books: []
};

export default Connect(Books);
export const Component = Books;
