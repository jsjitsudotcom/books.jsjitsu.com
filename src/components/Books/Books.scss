:local(.books) {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 0 10px;
  margin-top: 20px;
}

:local(.book){
  padding-bottom: 10px;
}

:local(.message){
  height: calc(100vh - 100px);
  display: flex;
  align-items: center;
  padding: 0 20px;
  font-size: 16px;
  color: rgba(black, 0.5);
  text-align: center;
  line-height: 1.5em;
  font-weight: 500;
}