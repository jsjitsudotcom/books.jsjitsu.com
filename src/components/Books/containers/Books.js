import { connect } from "react-redux";
import actions from "./../../../actions";

const mapState = state => ({
  books: state.books.books,
  loading: state.books.fetching
});

const mapDispatch = dispatch => ({
  onMore: () => dispatch(actions.books.fetchMore())
});

export default connect(mapState, mapDispatch);
