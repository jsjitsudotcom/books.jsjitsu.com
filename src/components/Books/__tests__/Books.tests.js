import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import { Component as Books } from "./../Books";

import booksData from "./../../../../__fixtures__/books-normalized.json";

describe("<Books />", () => {
  it("Le componsant doit être rendu", () => {
    shallow(<Books />);
  });

  describe("Suite de tests de l'affichage", () => {
    test("Les livres doivent bien s'afficher", () => {
      const wrapper = shallow(<Books books={booksData} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Suite de tests des methodes", () => {
    test("La méthode onMore doit être appelée", () => {
      const spy = sinon.spy();
      const wrapper = shallow(<Books books={booksData} onMore={spy} />);

      wrapper.find(".more").simulate("click");
      expect(spy.called).toEqual(true);
    });
  });
});
