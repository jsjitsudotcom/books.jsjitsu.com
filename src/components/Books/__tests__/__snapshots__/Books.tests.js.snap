// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Books /> Suite de tests de l'affichage Les livres doivent bien s'afficher 1`] = `
<div
  className="container"
>
  <div
    className="books"
  >
    <div
      className="book"
      key="0"
    >
      <Book
        onClick={[Function]}
        title="The Lord of the Rings"
      />
    </div>
    <div
      className="book"
      key="1"
    >
      <Book
        onClick={[Function]}
        title="Lord of the Rings"
      />
    </div>
    <div
      className="book"
      key="2"
    >
      <Book
        onClick={[Function]}
        title="Lords of the Ring"
      />
    </div>
    <div
      className="book"
      key="3"
    >
      <Book
        onClick={[Function]}
        title="Lord of the Rings"
      />
    </div>
  </div>
</div>
`;
